extractor/src/main/java/org/schabi/newpipe/extractor/utils/StringUtils.java

```diff
@@ -0,0 +1,49 @@
+package org.schabi.newpipe.extractor.utils;
+
+
+import javax.annotation.Nonnull;
+
+public class StringUtils {
+
+    private StringUtils() {
+    }
+
+    /**
+     * @param string The string to search in
+     * @param start A string from which to start searching.
+     * @return A substring where each '{' matches a '}'
+     * @throws IndexOutOfBoundsException If {@code string} does not contain {@code start}
+     */
+    @Nonnull
+    public static String matchToClosingParenthesis(@Nonnull final String string, @Nonnull final String start) {
+        int startIndex = string.indexOf(start);
+        if (startIndex < 0) {
+            throw new IndexOutOfBoundsException();
+        }
+
+        startIndex += start.length();
+        int endIndex = startIndex;
+        while (string.charAt(endIndex) != '{') {
+            ++endIndex;
+        }
+        ++endIndex;
+
+        int openParenthesis = 1;
+        int length = string.length();
+        while (openParenthesis > 0 && endIndex < length) {
```

Comment:
If you added `endIndex < length` here you should also add it to `while (string.charAt(endIndex) != '{') {` and to the `++endIndex;` three lines below. I left index checks out on purpose because an `IndexOutOfBounds` exception is better than returning a wrongly matched string that would anyway create exceptions down the line (e.g. when interpreting JS).

Reply:
I removed it. makes more sense this way, because then the catch block is used